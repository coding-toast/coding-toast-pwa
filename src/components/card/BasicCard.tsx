import * as React from 'react';
import { Card } from 'react-bootstrap';
import Link from 'next/link';

interface ICardProps {
  banner: string;
  description: string;
  publishDate: string;
  title: string;
}

const BasicCard: React.FC<ICardProps> = (props) => {
  const { title, banner, description, publishDate } = props;
  return (
    <Card style={{ margin: '16px 0', backgroundColor: 'rgba(255, 255, 255, 0.05)' }}>
      <Card.Img variant='top' src={banner} alt={`Banner ${title}`} style={{ minHeight: '275px', maxHeight: '275px' }} />
      <Card.Body>
        <Card.Title>{title}</Card.Title>
        <Card.Subtitle className='mb-2 text-style-muted text-color-secondary'>{publishDate}</Card.Subtitle>
        <Card.Text className='text-color-primary'>{description}</Card.Text>
        <Card.Link>
          <Link href='/'>
            <a>Read More</a>
          </Link>
        </Card.Link>
      </Card.Body>
    </Card>
  );
};

export default BasicCard;
