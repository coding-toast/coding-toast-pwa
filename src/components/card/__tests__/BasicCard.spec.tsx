import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BasicCard from '../BasicCard';

describe('<BasicCard />', () => {
  it('renders correctly', () => {
    const component = shallow(<BasicCard key='1' title='test' banner='test' description='test' publishDate='6/5/2020' />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
