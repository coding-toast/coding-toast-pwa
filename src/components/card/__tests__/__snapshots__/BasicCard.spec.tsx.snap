// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicCard /> renders correctly 1`] = `
<Card
  body={false}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.05)",
      "margin": "16px 0",
    }
  }
>
  <CardImg
    alt="Banner test"
    src="test"
    style={
      Object {
        "maxHeight": "275px",
        "minHeight": "275px",
      }
    }
    variant="top"
  />
  <CardBody>
    <CardTitle>
      test
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-style-muted text-color-secondary"
    >
      6/5/2020
    </CardSubtitle>
    <CardText
      className="text-color-primary"
    >
      test
    </CardText>
    <CardLink>
      <Link
        href="/"
      >
        <a>
          Read More
        </a>
      </Link>
    </CardLink>
  </CardBody>
</Card>
`;
